src/client/datascience/jupyter/jupyterSession.ts

```diff
@@ -82,40 +72,34 @@ export class JupyterSession extends BaseJupyterSession {
     }
 
     public async createNewKernelSession(
-        resource: Resource,
-        kernelConnection: KernelConnectionMetadata | undefined,
-        timeoutMS: number,
```

Comment:
All of this information is specific to this class (JupyterSession), never changing, hence passed into ctor and we have them in properties.